src/test/kotlin/com/example/flashcardbackend/integrationtest/DeckGroupIntegrationTest.kt

```diff
@@ -0,0 +1,338 @@
+                requestBuilder.findById(deckGroupLang.id)
+                    .andExpect(
+                        content().json(objectMapper.writeValueAsString(deckGroupLang)),
+                    )
+            }
+
+            @Test
+            @DisplayName("Should return the information of found deck group with empty deck")
+            fun shouldReturnFoundDeckGroupWithEmptyDeck() {
+                requestBuilder.findById(deckGroupMovies.id)
+                    .andExpect(jsonPath("$.id").value(deckGroupMovies.id))
+                    .andExpect(jsonPath("$.name").value(deckGroupMovies.name))
+                    .andExpect(jsonPath("$.description").value(deckGroupMovies.description))
+                    .andExpect(jsonPath("$.decks").isEmpty)
+            }
+        }
+
+        @Nested
+        @DisplayName("When deck group with given ID does not exist")
+        inner class WhenDeckGroupDoesNotExist {
+            private val id = 12
+
+            @Test
+            @DisplayName("Should return the HTTP status code NOT FOUND")
+            fun shouldReturnHttpStatusCodeNotFound() {
+                requestBuilder.findById(id)
+                    .andExpect(status().isNotFound)
+            }
+
+            @Test
+            @DisplayName("Should return a JSON error message")
+            fun shouldReturnJsonErrorMessage() {
+                requestBuilder.findById(id)
+                    .andExpect(jsonPath("$.message").value("Deck group with id $id not found."))
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("Create deck group")
+    inner class CreateDeckGroup {
+        @Nested
+        @DisplayName("When deck group is created successfully")
+        inner class WhenDeckGroupIsCreatedSuccessfully {
+            private val requestBody = """{"name": "Finnished", "description": "Learn Finnish with Finnished"}"""
+            private val deckGroupCreate = DeckGroupCreate("Finnished", "Learn Finnish with Finnished")
+
+            @Test
+            @DisplayName("Should return the HTTP status code CREATED")
+            fun shouldReturnHttpStatusCodeCreated() {
+                requestBuilder.createDeckGroup(requestBody)
+                    .andExpect(status().isCreated)
+            }
+
+            @Test
+            @DisplayName("Should insert the deck group in the repository")
+            fun shouldInsertDeckGroupInTheRepository() {
+                requestBuilder.createDeckGroup(requestBody)
+                // Verify the last deck group is the newly created one
+                requestBuilder.findAll()
+                    .andExpect(jsonPath(("$"), Matchers.hasSize<Int>(4)))
+                    .andExpect(jsonPath("$[3].name").value(deckGroupCreate.name))
+                    .andExpect(jsonPath("$[3].description").value(deckGroupCreate.description))
+            }
+        }
+
+        @Nested
+        @DisplayName("When deck group name is blank")
+        inner class WhenDeckGroupNameIsBlank {
```

Comment:
You could modify your request builder to a take a DTO as a method parameter and transform the DTO into JSON by using object mapper.

Reply:
Will fix (maybe tomorrow) :)